src/dns/remote.rs

```diff
@@ -0,0 +1,113 @@
+use crate::dns::{DnsEntry, DnsResolver, ResolveType};
+use crate::types;
+use core::str::FromStr;
+use hickory_resolver::config::{ResolverConfig, ResolverOpts};
+use hickory_resolver::Resolver;
+use log::trace;
+use std::net::IpAddr;
+
+pub struct RemoteResolver {
+    hickory: Resolver,
+}
+
+impl DnsResolver for RemoteResolver {
+    fn resolve(
+        &mut self,
+        domain: &str,
+        resolve_type: ResolveType,
+    ) -> Result<DnsEntry, types::Error> {
+        let mut entry = DnsEntry::new(domain, vec![]);
+
+        let mut ip_types = Vec::new();
+        match resolve_type {
+            ResolveType::Ipv4 => {
+                ip_types.push("ipv4");
+            }
+            ResolveType::Ipv6 => {
+                ip_types.push("ipv6");
+            }
+            ResolveType::Both => {
+                ip_types.push("ipv6");
+                ip_types.push("ipv4");
```

Comment:
It strikes me as unusual that we're using a static string here as a type marker.

Reply:
I guess we can just use the ResolverType itself and if it is Both we Push `ResolverType::Ipv4` and `ReolverType::Ipv6`

Reply:
👍